### PR TITLE
[KYUUBI #2741] Add kyuubi-spark-connector-common module

### LINKED ISSUE #2741: [Improvement] We need the kyuubi-spark-connector-common module to contain things that are commonly used by multiple spark connectors
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

like SparkUtils LocalSparkSession

### How should we improve?

like SparkUtils LocalSparkSession

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #2741 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by jiaoqingbo** [comment]
Can anyone help me to see why the Style check fails, do I need to manually deploy spark-connector-common to the remote repository？

**Comment by jiaoqingbo** [comment]
cc @yaooqinn @pan3793

File: pom.xml [hunk 1]
```diff
@@ -70,6 +70,7 @@
         <module>extensions/spark/kyuubi-spark-authz</module>
         <module>extensions/spark/kyuubi-spark-connector-tpcds</module>
         <module>extensions/spark/kyuubi-spark-connector-tpch</module>
+        <module>extensions/spark/kyuubi-spark-connector-common</module>
```
Comment (pan3793) [inline pom.xml:73]: Sort alphabetically

**Review by pan3793 — APPROVED** [review]
LGTM, except two minor comments

File: extensions/spark/kyuubi-spark-connector-common/pom.xml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <configuration>
+                    <skip>true</skip>
+                </configuration>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (jiaoqingbo) [reply extensions/spark/kyuubi-spark-connector-common/pom.xml:99]: I will remove this

File: extensions/spark/kyuubi-spark-connector-common/pom.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-sql_${scala.binary.version}</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-connector-common/pom.xml:54]: nit: one empty line

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**